### ISSUE TITLE
🐛 Nested biome.json files in monorepo not detected

### ISSUE THREAD

**Issue by ozziexsh** [biomejs/biome#972]
### Environment information

```block
CLI:
  Version:                      1.4.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v16.14.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/8.3.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

I have created a repo here that reproduces the error

https://github.com/ozziexsh/biome-monorepo-error

I have a root `biome.json` file which has a rule `"noUndeclaredVariables": "error"`

and a `packages/one/biome.json` that has 

```
"extends": ["../../biome.json"],
"javascript": {
  "globals": ["someGlobal"]
}
```

which should allow a global variable only in the `packages/one` folder. The VSCode extension highlights this global as undeclared, the same output happens while running the lint command via the cli.

If you add the global to the root `biome.json` the error goes away.

### Expected result

The `packages/one/biome.json` file should be respected in the `packages/one` folder

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ozziexsh** [comment]
Yes, the same output happens while running the lint command via the cli

**Comment by saturn30** [comment]
It is the same in version 1.5.3.

**Comment by ematipico** [comment]
This should be fixed by https://github.com/biomejs/biome/pull/1700

**Comment by ematipico** [comment]
This was stale for a while, sorry.

Looking into it, and I don't see any script inside the reproduction, which makes me wonder where the CLI is used. The root? Somewhere else?

Plus, Biome doesn't support multiple `biome.json` files nor nested configuration files.

In a monorepo you want to have one `biome.json`, and use `overrides` to change its behaviour in certain folders.

**Comment by Igosuki** [comment]
Will biome support nested configs like OP outlined ?

**Comment by ematipico** [comment]
> Will biome support nested configs like OP outlined ?

I'd like to understand why there's a need for nested configuration files, unless that's for workspaces.

**Comment by daegren** [comment]
For me, the need is that I'm trying to share a common `biome.json` config among internal projects inside a turbo monorepo. But also expose it as an NPM package for other projects to be able to consume it.

e.g.

```json
{
  "$schema": "https://biomejs.dev/schemas/1.5.3/schema.json",
  "extends": ["@my-namespace/biome-config/index.json"]
}
```

**Comment by ozziexsh** [comment]
> I'd like to understand why there's a need for nested configuration files, unless that's for workspaces.

Yes we are using yarn workspaces, e.g. `packages/client` and `packages/server`. Client has some config that needs to be different from server, and some that should be shared between the two.

> Plus, Biome doesn't support multiple biome.json files nor nested configuration files.

It's in the docs though? https://biomejs.dev/guides/big-projects/#share-the-configuration

**Comment by ematipico** [comment]
Sorry @ozziexsh, I actually mean LSP workspaces, which have a different semantic compared to the workspaces of a package manager.

> It's in the docs though?

I can see the confusion/misunderstanding now. I suppose the docs need to clarify the use case.

The main idea behind the part of the docs is that a team has a project, another team has another project. 

I'll have to review that chapter.

**Comment by Igosuki** [comment]
I have a specific use case here, I use biome in a monorepo, where there is a an API app and a front end app, the API app enforces a rule to prevent using the process.env global outside of a few files, because the api is configured by a single package file.
However, the front app uses nextjs, and it has its own idiomatic rules, I find using overrides at the root a bit cumbersome.

**Comment by devdumpling** [comment]
+1 this -- I have a similar setup (turborepo backed monorepo). I'm testing migrating it to biome. Currently I have a shared eslint config defined in a `@company/eslint-config` package that exports the shared config. I then consume this in other repos via the package.

It's not the end of the world to have a root `biome.json` as described in the docs, but would love to be able to extend from a package as opposed to a path, e.g. `extends: ["@company/config.biome"]`

**Comment by ematipico** [comment]
> It's not the end of the world to have a root biome.json as described in the docs, but would love to be able to extend from a package as opposed to a path, e.g. extends: ["@company/config.biome"]

@devdumpling you'll be able to this in the next version

**Comment by castarco** [comment]
> Closed by #1772

@ematipico Does it solve this specific issue? As I understand, #1772 is about resolving "packaged" configurations in the `extends` field, but it does not cover the case of applying nested configurations when `biome` runs from the root directory.

Not sure if this was closed by confusion due to the last conversation bits of this thread.

**Comment by guillempuche** [comment]
Still in version 1.6.4 doesn't work for me.

For example, this config doesn't prevent VSCode of showing me errors, instead of warnings or disabled.

// packages/package_a/biome.json
```json
{
	"extends": [
		"../../../biome.json"
	],
	"linter": {
		"rules": {
			"complexity": {
				"noStaticOnlyClass": "off"
			},
			"suspicious": {
				"noExplicitAny": "warn",
				"noShadowRestrictedNames": "off"
			}
		}
	}
}
```

**Comment by ematipico** [comment]
Nesting configuration files are a known limitation. You can track progresses and/or help in #2228 

As a workaround, I suggest you to always have a `biome.json` file at the root of you project, and use `include`/ignore`/`overrides` to identify the folders that need to be checked.
